evennia/contrib/auditing.py

```diff
@@ -0,0 +1,167 @@
+"""
+
+Contribution - Johnny 2017
+
+This is an extension of the base ServerSession class that parses
+all server inputs and/or outputs and passes a dict containing the parsed
+metadata to a method of your choosing.
+
+This is useful for recording player activity where necessary for auditing,
+usage data or post-incident forensic analysis.
+
+Somewhere in your game (or even right there in settings.py) should be a 
+method that receives dict objects and does something with them, whether 
+writing to a database, a file, or some external queue. A simplistic example 
+might be something as brief as:
+
+from django.utils import timezone
+from evennia.utils.logger import log_file
+import json
+
+def just_write_it(data, **kwargs):
+    # Bucket logs by day
+    bucket = timezone.now().strftime('%Y%m%d')
+    # Remove the objects; keep the values
+    data.pop('objects')
+    # Write it
+    log_file(json.dumps(data), filename="auditing_%s.log" % bucket)
+
+The parsed metadata contains the account name and ID, the character name 
+and ID, the room name and ID, and the IP address of the connected session
+issuing the command. The respective objects are also passed in the event
+you wish to store data back to them or manipulate them in some other way.
+
+*** WARNING ***
+All strings are recorded and stored in plaintext. This includes those strings
+which might contain sensitive data (create, connect, password).
+
+You are expected to implement proper masking/scrubbing of these and any
+other commands upon receipt and storing the recorded data securely. Please do
+not store user passwords in plaintext, and be ethical-- this records even
+private communications.
+***************
+
+Installation:
+
+Designate this class as the SERVER_SESSION_CLASS in `settings.py`.
+
+i.e. SERVER_SESSION_CLASS = "evennia.contrib.auditing.AuditedServerSession"
+
+There are three additional settings that govern operation-- whether you
+want to record inbound strings and/or outbound strings, and what method
+you want the data sent to.
+
+AUDIT_METHOD = just_write_it
+AUDIT_IN = True or False
+AUDIT_OUT = True or False
+
+Recording outbound strings can be noisy-- a single message broadcast to 12
+occupants of a room will generate 12 of the same message, one to each.
+"""
+
+import os
+import socket
+
+from django.utils import timezone
+from evennia import settings as ev_settings
+from evennia.utils import logger
+from evennia.server.serversession import ServerSession
+
+class AuditedServerSession(ServerSession):
+    def __init__(self, *args, **kwargs):
+        super(AuditedServerSession, self).__init__(*args, **kwargs)
+        
+        # Attributes governing auditing of commands and where to send logs
+        self.audit_method = getattr(ev_settings, 'AUDIT_METHOD', None)
```

Comment:
This could be loaded as global variables up top, no need to have it in the `__init__`.